src/mlpack/methods/ann/layer/multihead_attention_impl.hpp

```diff
@@ -97,14 +97,15 @@ Forward(const MatType& input, MatType& output)
   // The shape of q : (embedDim, tgtSeqLen, batchSize).
   // The shape of k : (embedDim, srcSeqLen, batchSize).
   // The shape of v : (embedDim, srcSeqLen, batchSize).
-  const CubeType q(const_cast<MatType&>(input).memptr(),
-      embedDim, tgtSeqLen, batchSize, false, false);
-  const CubeType k(const_cast<MatType&>(input).memptr() +
-      (selfAttention ? 0 : embedDim * tgtSeqLen * batchSize),
-      embedDim, srcSeqLen, batchSize, false, false);
-  const CubeType v(const_cast<MatType&>(input).memptr() +
+  CubeType q;
+  MakeAlias(q, input, embedDim, tgtSeqLen, batchSize, 0, false);
+  CubeType k;
+  MakeAlias(k, input, embedDim, srcSeqLen, batchSize,
+      (selfAttention ? 0 : embedDim * tgtSeqLen * batchSize), false);
+  CubeType v;
```

Comment:
```suggestion
  CubeType q, k, v;
  MakeAlias(q, input, embedDim, tgtSeqLen, batchSize, 0, false);
  MakeAlias(k, input, embedDim, srcSeqLen, batchSize,
      (selfAttention ? 0 : embedDim * tgtSeqLen * batchSize), false);
```

Up to you, maybe this is cleaner in your eyes?